src/Bishop.py

```diff
@@ -1,6 +1,14 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterator
```

Comment:
`typing.Iterator` is deprecated. please use `collections.abc.Iterator` instead.

Reply:
I agree on `list`, I missed that, but are you sure on the `Iterator`? I can change it, but https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#functions says `typing` specifically.

Reply:
Yeah, you're right. So leave the `typing.Iterator` as it is. I think it won't make too much work when we have to switch to the other thing.